packages/web/pages/user/my-account/my-orders.js

```diff
@@ -0,0 +1,348 @@
+								/>
+							) : (
+								<NoOrders>{t('account:messages.noOrdersToShow')}</NoOrders>
+							)}
+						</MainContent>
+					</MainContentContainer>
+				)}
+			</Protected>
+		</Container>
+	);
+};
+
+MyOrders.propTypes = {
+	orders: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+	currentPage: PropTypes.number.isRequired,
+	totalPages: PropTypes.number.isRequired,
+	totalItems: PropTypes.number.isRequired,
+	currentSort: PropTypes.shape({
+		by: PropTypes.string,
+		order: PropTypes.string,
+	}),
+};
+
+MyOrders.defaultProps = {
+	currentSort: {},
+};
+
+MyOrders.getInitialProps = async (ctx) => {
+	const { query } = ctx;
+
+	const page = Number(query.page) || 1;
+
+	return {
+		orders: ordersMock,
+		currentPage: page,
+		totalPages: 1,
+		totalItems: 3,
+		currentSort: { by: query.orderBy, order: query.order },
+		sortOptions,
+		namespacesRequired: ['helper', 'account', 'profile', 'datagrid'],
+	};
+};
+
+export const Container = styled.div`
+	display: flex;
+	margin: 0 auto;
+	background-color: ${({ theme }) => theme.colors.whiteSmoke};
+	padding: 3rem 4rem 6rem;
+
+	> section:first-child {
+		margin-right: 4rem;
+	}
+
+	@media screen and (max-width: 950px) {
+		flex-direction: column;
+
+		> section:first-child {
+			margin-bottom: 1rem;
+		}
+	}
+`;
+
+export const MainContentContainer = styled.section`
+	width: 100%;
+`;
+
+export const MainContent = styled.div`
+	min-height: 80vh;
+	background-color: ${({ theme }) => theme.colors.white};
+	padding: 2rem;
+`;
+
+export const InfoContainer = styled.div`
+	display: flex;
+	align-items: center;
+	justify-content: space-between;
+	margin-bottom: 1rem;
+
+	@media screen and (max-width: 950px) {
+		flex-direction: column;
+
+		button {
+			margin-bottom: 1rem;
+		}
+	}
+`;
+
+export const NoOrders = styled.span`
+	color: ${({ theme }) => theme.colors.darkGray};
+	font-size: 2rem;
+`;
+
+const statusModifiers = {
+	dealStruck: (colors) => css`
+		color: ${colors.secondary};
+		&::before {
+			background: ${colors.secondary};
+		}
+	`,
+	dealOngoing: (colors) => css`
+		color: ${colors.lightGray2};
+		&::before {
+			background: ${colors.lightGray2};
+		}
+	`,
+	dealCancelled: (colors) => css`
+		color: ${colors.red};
+		&::before {
+			background: ${colors.red};
+		}
+	`,
+};
+
+export const DealStatus = styled.div`
+	${({ theme: { colors }, status }) => css`
+		display: inline-block;
+		position: relative;
+		line-height: 2.4rem;
+		font-weight: 500;
+		padding: 0.2rem 0.8rem;
+		max-width: fit-content;
+		text-align: center;
+
+		&::before {
+			content: '';
+			display: block;
+			position: absolute;
+			top: 0;
+			left: 0;
+			width: 100%;
+			height: 100%;
+			border-radius: 1.45rem;
+			opacity: 0.1;
+		}
+
+		${status === dealStatusEnum.DEAL_STRUCK && statusModifiers.dealStruck(colors)};
+		${status === dealStatusEnum.DEAL_ONGOING && statusModifiers.dealOngoing(colors)};
+		${status === dealStatusEnum.DEAL_CANCELLED && statusModifiers.dealCancelled(colors)};
```

Comment:
```suggestion
		${statusModifiers[dealStatusEnum.DEAL_STRUCK](colors)};
```

Reply:
See below comment

Reply:
Sorry! The correct code would look like this:
```suggestion
		${statusModifiers[status](colors)};
```

Reply:
Interesting 🤔